Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -932,13 +961,68 @@ public static void CreateUpdateRunSetFromJSON(RunsetExecutor runsetExecutor, Gin
                                     throw new Exception(error);
                                 }
                             }
+                            /// For Virtual Business Flows which either exists in solution or are to be created for execution only
+                            /// Case : Virtual Runset, Runners & Business Flows
+                            /// Case : Virtual / Non-Virtual Runset, Runners with Business Flows that exist in the solution 
+                            ///        but are not part of current Runset & Runners
                             else
                             {
+                                try
+                                {
+                                    /// Business Flow exist in the Solution but isn't already part of Runner/Runset
+                                    /// Fetch BF & use the same for execution
+                                    if (businessFlowConfig.ID != null || !string.IsNullOrEmpty(businessFlowConfig.Name))
+                                    {
+                                        bf = (BusinessFlow)FindItemByIDAndName<BusinessFlow>(
+                                         new Tuple<string, Guid?>(nameof(BusinessFlow.Guid), businessFlowConfig.ID),
+                                         new Tuple<string, string>(nameof(BusinessFlow.Name), businessFlowConfig.Name),
+                                         WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<BusinessFlow>());
+                                    }
+                                }
+                                catch(Exception exc)
```

Comment:
no need this try-catch, if ID was not found so exception needs to be thrown and execution to fail like we having for any other entity ID in the JSON

Reply:
This is for creating Virtual Business Flows which doesn't exists in the solution

Reply:
we agreed that no need to create virtual if ID provided but not found, need to throw exception

Reply:
This is outdated changes Meni. In the latest commit, it is fixed @ https://github.com/Ginger-Automation/Ginger/pull/2283/commits/766ecd5f48abad5a24b5fc531fd39eee4c68bc1d

![image](https://user-images.githubusercontent.com/45287388/154898370-6a5177d9-016e-486b-b1c1-ceba534285ed.png)